Added ESC FAQs (#3701)
pulumi/pulumi-hugo: themes/default/content/docs/pulumi-cloud/esc/faq.md

---
title_tag: "Pulumi ESC FAQs"

> meta_desc: Frequently asked questions about preview, GA, pricing and roadmap 

Reviewer: ```suggestion
meta_desc: Frequently asked questions about Pulumi ESC preview, GA, pricing and roadmap.
```

title: FAQ
h1: Pulumi ESC FAQs
meta_image: /images/docs/meta-images/docs-meta.png
menu:
  pulumicloud:
    parent: esc
    weight: 6
    identifier: faq
---

> **Is it safe to use Pulumi ESC for my production stacks?**

Reviewer 2: This questions makes me feel like the answer should be less about HA and more about security. Is it worth talking about the elements of security built into this service? Given that this is already discussed on the Pulumi Security Whitepaper, it may be worth linking to that at a minimum. I would still make a mention of something about security here like the following:

Yes, Pulumi ESC is served using [the same highly secure, highly available infrastructure](/security/pulumi-cloud-security-whitepaper.pdf) as the rest of Pulumi Cloud...

Author: updated! Thank you.

Author: updated!

> **Is it safe to use Pulumi ESC for my production stacks?**
> <br>

Reviewer: ```suggestion
## Is it safe to use Pulumi ESC for my production stacks?

```

> Yes, Pulumi ESC is served using the same high-availability infrastructure as the rest of Pulumi Cloud. Preview denotes that we are rapidly iterating and improving the feature set. It is unlikely that we will make any breaking changes. Should we introduce a new API, we are dedicated to ensuring optimal compatibility to prevent any interruptions to your infrastructure.

Reviewer 2: ```suggestion
Yes, Pulumi ESC is served using the same high-availability infrastructure as the rest of Pulumi Cloud. This service is currently in preview, meaning that we are rapidly iterating and improving its feature set. It is unlikely that we will make any breaking changes. Should we introduce a new API, we are dedicated to ensuring optimal compatibility to prevent any interruptions to your infrastructure.
```

> Yes, Pulumi ESC is served using the same [highly secure](https://www.pulumi.com/security/pulumi-cloud-security-whitepaper.pdf) and highly available infrastructure as the rest of Pulumi Cloud. The service is currently in preview as we are rapidly iterating and improving the feature set. It is unlikely that we will make any breaking changes. Should we introduce a new API, we are dedicated to ensuring optimal compatibility to prevent any interruptions to your infrastructure.

Reviewer 3: ```suggestion
Yes, Pulumi ESC is served using the same [highly secure](/security/pulumi-cloud-security-whitepaper.pdf) and highly available infrastructure as the rest of Pulumi Cloud. The service is currently in preview as we are rapidly iterating and improving the feature set. It is unlikely that we will make any breaking changes. Should we introduce a new API, we are dedicated to ensuring optimal compatibility to prevent any interruptions to your infrastructure.
```

> **Why did Pulumi launch ESC?**
> <br>

Reviewer: ```suggestion
## Why did Pulumi launch ESC?

```

> We launched Pulumi ESC in response to customer feedback about their difficulties in managing config and secrets causing sprawl and duplications across stacks. Pulumi ESC is a natural extension of Pulumi IaC by allowing [hierarchical](http://localhost:1313/docs/pulumi-cloud/esc/#configuration-as-code) configurations that can be used across stacks.

Reviewer 2: ```suggestion
We launched Pulumi ESC in response to customer feedback about their difficulties in managing configuration and secrets across multiple environments, causing sprawl and duplications across stacks. Pulumi ESC is a natural extension of Pulumi IaC by allowing [hierarchical](/docs/pulumi-cloud/esc/#configuration-as-code) configurations that can be used across stacks.
```

Author: explicity called out config because of the reference to pulumi config files..

> We launched Pulumi ESC in response to customer feedback about their difficulties in managing config and secrets causing sprawl and duplications across stacks. Pulumi ESC is a natural extension of Pulumi IaC by allowing [hierarchical](http://localhost:1313/docs/pulumi-cloud/esc/#configuration-as-code) configurations that can be used across stacks.

Reviewer 3: ```suggestion
We launched Pulumi ESC in response to customer feedback about their difficulties in managing config and secrets causing sprawl and duplications across stacks. Pulumi ESC is a natural extension of Pulumi IaC by allowing [hierarchical](/docs/pulumi-cloud/esc/#configuration-as-code) configurations that can be used across stacks.
```

> **When will Pulumi ESC be generally available?**
> <br>

Reviewer: ```suggestion
## When will Pulumi ESC be generally available?

```

Pulumi ESC will be generally available in 2024.

> **What will be the pricing of Pulumi ESC?**
> <br>

Reviewer: ```suggestion
## What will be the pricing of Pulumi ESC?

```

During the preview, the product is free to use. When the product is made generally available, it will have a pricing model that is competitive with the market.

**What is the roadmap for Pulumi ESC?**
<br>
We have a lot of [features](https://github.com/pulumi/esc/issues) planned for ESC. Here are a few highlights:

- [Webhooks](https://github.com/pulumi/esc/issues/188)
- [Syncing config/secrets to external systems](https://github.com/pulumi/esc/issues/58)
- [Version controlled environments](https://github.com/pulumi/esc/issues/63)
- [Table UI authoring experience](https://github.com/pulumi/esc/issues/62)

We encourage you to create a [feature request](https://github.com/pulumi/esc/issues/new/choose) or upvote [existing open requests](https://github.com/pulumi/esc/issues).

> **Can a Self-hosted Pulumi customer use this?**
> <br>

Reviewer: ```suggestion
## Can a Self-hosted Pulumi customer use this?

```

> Pulumi ESC is currently not supported for self-hosted customers. [Contact sales](https://www.pulumi.com/contact/?form=sales) if you'd like to participate in our self-hosted beta for ESC when it is available

Reviewer 2: ```suggestion
Pulumi ESC is currently not supported for self-hosted customers. [Contact sales](https://www.pulumi.com/contact/?form=sales) if you'd like to participate in our self-hosted beta for ESC when it is available.
```

> Pulumi ESC is currently not supported for self-hosted customers. [Contact sales](https://www.pulumi.com/contact/?form=sales) if you'd like to participate in our self-hosted beta for ESC when it is available.

Reviewer: ```suggestion
Pulumi ESC is currently not supported for self-hosted customers. [Contact sales](/contact/?form=sales) if you'd like to participate in our self-hosted beta for ESC when it is available.
```
